### DISCUSSION TITLE
How can I pass data while calling forge script?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [foundry-rs/foundry#2456] [Help]
I'm trying to invoke the `run()` method and pass a parameter, how can I do that? From the document the default method is `run()` but it doesn't tell me how to pass a parameter. 

```
forge script --sender XXX -vvv --private-key XXX --fork-url https://rinkeby.infura.io/v3/XXX script/Contract.s.sol
```

Here's the test contract:

```
contract TestContract {
     function run(address _addr) public {}
}
```

**Comment by burnto** [comment]
I am currently using environment variables for this, but it's kind of ugly:

```solidity
address owner = vm.envAddress("OWNER");
uint256 pk = vm.envOr("DEPLOYER_PRIVATE_KEY", uint256(0));
pk == 0 ? vm.startBroadcast() : vm.startBroadcast(pk);
/// …
```

The `forge script --help` docs indicate that `ARGS` can be passed, but I haven't figured out if/how this works yet.

**Comment by mds1** [comment]
You can do this similar to how to you'd pass arguments to `cast call`:

```solidity
contract MyScript is Script {
    function setUp() public {}

    function run(uint256 x) public {
        console2.log(x);
    }
}
```

Then invoke with `forge script script/MyScript.s.sol --sig "run(uint256)" 5`

**Reply by burnto** [reply]
lovely, thank you!
